src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -343,7 +343,7 @@ def __init__(
         self.account_group_regex = account_group_regex
         self.workspace_group_regex = workspace_group_regex
```

Comment:
If the caller is allowed to pass these in as strings (instead of compiled patterns) I'd really like to compile them here during isn't initialisation. This isn't (just) a performance thing: a) it catches problems with regex syntax (a common problem) during initialisation rather than later when we use it for the first time; and b) any time a string is converted to its domain type lots of things just become simpler (IDE completions, type checking, etc.)

Reply:
Done, note that I kept the public API the same